### PR TITLE
Add example to use internal realms

### PR DESCRIPTION
**Pull request by pickypg**
Allows the user to decide between using internal realms or not.

### CODE REVIEW

**Comment by pickypg** [comment]
cc @jaymode

File: src/main/java/org/elasticsearch/example/realm/CustomAuthenticationFailureHandler.java [hunk 1]
```diff
@@ -68,10 +68,32 @@ public ElasticsearchSecurityException missingToken(RestRequest request) {
     }
 
     @Override
-    public ElasticsearchSecurityException missingToken(TransportMessage message, String action) {
```
Comment (jaymode) [inline src/main/java/org/elasticsearch/example/realm/CustomAuthenticationFailureHandler.java]: looks like you replaced the wrong version of this method, should have replaced the one taking a RestRequest

Comment (pickypg) [inline src/main/java/org/elasticsearch/example/realm/CustomAuthenticationFailureHandler.java]: Yep. I noticed too. :) Rebased and fixed.

File: src/main/java/org/elasticsearch/example/realm/CustomAuthenticationFailureHandler.java [hunk 2]
```diff
@@ -98,4 +96,36 @@ public ElasticsearchSecurityException authenticationRequired(String action) {
         se.addHeader("WWW-Authenticate", "custom-challenge");
         return se;
     }
+
+    /**
+     * Allows you to allow the internal realms (or even other custom realms)
```
Comment (jaymode) [inline src/main/java/org/elasticsearch/example/realm/CustomAuthenticationFailureHandler.java]: Let's reword this a bit:

```
This method handles the logic for adding a custom header when there is a missing token, which includes the ability to preserve existing authenticate challenges (WWW-Authenticate headers) or to overwrite them
```

There is only one failure handler, so there will be no other custom realms that add these headers

**Comment by jaymode** [comment]
left a small doc comment, OTT LGTM

**Pull request closed without merging**